### PR TITLE
Adds support for resource templating

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Adds support for templating using the new `configuration` package, and wires that into the current templating support.

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
See https://github.com/giantswarm/giantswarm/issues/1332 for tracking updating away from `%%DOCKER_TAG%%` - we keep backwards compatability with that for now

**Comment by JosephSalisbury** [comment]
https://github.com/giantswarm/api/issues/339 and https://github.com/giantswarm/g8s-prometheus/issues/7 are updates for `api` and `g8s-prometheus` that update their templating with regards to ingress tags

File: template/utils.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package template
```
Comment (xh3b4sd) [inline template/utils.go]: IMO `utils.go` implies that something is wrong with the mental model.

File: workflow/clusters.go [hunk 2]
```diff
@@ -1,26 +1,44 @@
 package workflow
 
-import "github.com/spf13/afero"
+import (
+	"github.com/giantswarm/architect/configuration"
+	"github.com/giantswarm/architect/installation"
+	"github.com/spf13/afero"
+)
+
+type KubernetesCluster struct {
```
Comment (JosephSalisbury) [inline workflow/clusters.go]: This has been moved here from `workflow.go`, I forgot about it during the previous refactor

File: workflow/clusters.go [hunk 3]
```diff
@@ -34,11 +52,13 @@ func ClustersFromEnv(fs afero.Fs, workingDirectory string) ([]KubernetesCluster,
 
 		newCluster := KubernetesCluster{
 			ApiServer:      configuredCluster.ApiServer,
-			IngressTag:     configuredCluster.IngressTag,
+			Prefix:         configuredCluster.EnvVarPrefix,
```
Comment (JosephSalisbury) [inline workflow/clusters.go]: We now use the env var prefix for the templated directory name

File: workflow/clusters_test.go [hunk 4]
```diff
@@ -0,0 +1,200 @@
+package workflow
```
Comment (JosephSalisbury) [inline workflow/clusters_test.go]: `clusters_test.go` is me fixing the fact i forgot to move this as well

File: workflow/workflow_test.go [hunk 5]
```diff
@@ -370,175 +366,3 @@ func TestGetDeployWorkflow(t *testing.T) {
 		}
 	}
 }
-
-func TestClustersFromEnv(t *testing.T) {
```
Comment (JosephSalisbury) [inline workflow/workflow_test.go]: Moved to `clusters_test.go`

**Pull request merged**